crypto/core_namemap.c

```diff
@@ -107,7 +107,8 @@ int ossl_namemap_empty(OSSL_NAMEMAP *namemap)
     if (namemap == NULL)
         return 1;
 
-    CRYPTO_THREAD_read_lock(namemap->lock);
+    if (!CRYPTO_THREAD_read_lock(namemap->lock))
+        return -1;
```

Comment:
Why are you returning -1 here? Why not just 0.
You might want to search for usages of ossl_namemap_empty.
i.e if (namemap != NULL && ossl_namemap_empty(namemap)) {

Reply:
I don't recall my reasoning, but it seems like 0/1 is only when you know the answer. Happy to change it if @paulidale (or other reviewer) doesn't mind.

Reply:
I'm torn both ways -- both have advantages and disadvantages.

Outside of tests, there is only one use of this function and it is checking for empty.  This and the function's Boolean sounding name lead me to think that 0 on error might be the better choice.  The documentation in internal/man3 will need to be updated to note this.

Reply:
You can also make ossl_namemap_stored() return NULL in case of error.

Reply:
additional commit pushed that returns 0 on error, fixed one place to handle that, and updated the docs in 82578da